webui/react/src/pages/TrialDetails.tsx

```diff
@@ -71,11 +72,21 @@ const trialContinueConfig = (
   };
 };
 
+const STORAGE_PATH = 'trial-detail';
+const STORAGE_CHECKPOINT_VALIDATION_KEY = 'checkpoint-validation';
+const STORAGE_METRICS_KEY = 'metrics';
+
 const TrialDetailsComp: React.FC = () => {
   const { trialId: trialIdParam } = useParams<Params>();
   const trialId = parseInt(trialIdParam);
+  const [ trialResponse, triggerTrialRequest ] =
+    useRestApi<TrialDetailsParams, TrialDetails>(getTrialDetails, { id: trialId });
   const [ experiment, setExperiment ] = useState<ExperimentDetails>();
-  const [ showHasCheckpoint, setShowHasCheckpoint ] = useState(true);
+  const storage = useStorage(STORAGE_PATH);
+  const trialStorage = useStorage(`${STORAGE_PATH}/${trialResponse?.data?.experimentId}`);
```

Comment:
we might need to make availability of the storage dependent on the experiment id.